src/scry/text_document.cr

```diff
@@ -81,6 +81,11 @@ module Scry
       @text.first
     end
 
+    def get_line(line_number)
+      lines = source.split("\n")
```

Comment:
That can be pretty bad for perf' and memory usage (especially for big files)

Maybe something with `source.each_line.skip(line_number - 1).next` ? This will still allocate a string for each line until it finds the line..